payments/schema/mutations.py

```diff
@@ -394,52 +394,67 @@ def mutate_and_get_payload(cls, root, info, **input):
         return CreateAdditionalProductOrderMutation(order=order)
 
 
+class UpdateOrderInput(OrderInput, graphene.InputObjectType):
+    id = graphene.ID(required=True)
+
+
 class UpdateOrderMutation(graphene.ClientIDMutation):
-    class Input(OrderInput):
-        id = graphene.ID(required=True)
+    class Input:
+        orders = graphene.List(UpdateOrderInput, required=True)
 
-    order = graphene.Field(OrderNode)
+    successful_orders = graphene.List(OrderNode)
+    failed_orders = graphene.List(FailedOrderType)
 
     @classmethod
     @change_permission_required(Order)
     @view_permission_required(BerthLease, WinterStorageLease)
-    @transaction.atomic
-    def mutate_and_get_payload(cls, root, info, **input):
-        order = get_node_from_global_id(
-            info, input.pop("id"), only_type=OrderNode, nullable=False
-        )
-        lease_id = input.pop("lease_id", None)
-        if lease_id:
-            lease = None
-            try:
-                lease = get_node_from_global_id(
-                    info, lease_id, BerthLeaseNode, nullable=True
-                )
-            # If a different node type is received get_node raises an assertion error
-            # when trying to validate the type
-            except AssertionError:
-                lease = get_node_from_global_id(
-                    info, lease_id, WinterStorageLeaseNode, nullable=True
-                )
-            finally:
-                if not lease:
-                    raise VenepaikkaGraphQLError(
-                        _("Lease with the given ID does not exist")
-                    )
-                input["lease"] = lease
+    def mutate_and_get_payload(cls, root, info, orders, **input):
 
-        try:
-            # handle case where input has both lease and status.
-            # set order status only after changing the lease, because setting order status
-            # usually triggers a change in lease status.
-            new_status = input.pop("status", None)
-            update_object(order, input)
-            if new_status:
-                order.set_status(new_status, _("Manually updated by admin"))
+        successful_orders = []
+        failed_orders = []
 
-        except (ValidationError, IntegrityError) as e:
-            raise VenepaikkaGraphQLError(e)
-        return UpdateOrderMutation(order=order)
+        for order_input in orders:
+            order_id = order_input.pop("id")
+            try:
+                with transaction.atomic():
+                    order = get_node_from_global_id(
+                        info, order_id, only_type=OrderNode, nullable=False
+                    )
+                    lease_id = order_input.pop("lease_id", None)
+                    if lease_id:
```

Comment:
This can now be simplified toÑ
```suggestion
                    if lease_id := order_input.pop("lease_id", None):
```